### ISSUE TITLE
problems with LSP client plugin

### ISSUE THREAD

**Issue by MurraySobol** [Ekopalypse/NppLspClient#25]
Your comment was: 
This sounds like Rust is not part of your Notepad++ setup because keyword coloring is done by Npp/Scintilla and is not part of the language server protocol and if that is the case, then the NppLspClient plugin will not work with Rust files.
Notepad++ debug info:
Notepad++ v8.7.4   (64-bit)
Build time : Dec  4 2024 - 23:50:05
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
Periodic Backup : ON
Placeholders : OFF
DirectWrite : OFF
Multi-instance Mode : monoInst
File Status Auto-Detection : cdEnabledNew (for current file/tab only)
Dark Mode : OFF
OS Name : Windows 11 Pro (64-bit)
OS Version : 24H2
OS Build : 26100.2454
Current ANSI codepage : 1252
Plugins : 
    mimeTools (3.1)
    NppConverter (4.6)
    NppExport (0.4)
    NppLspClient (0.0.31)
    rustnpp (1.0.2)

I also checked the langs.xml file for the Rust section, it is there: 
        </Language>
        <Language name="rust" ext="rs" commentLine="//" commentStart="/*" commentEnd="*/">
            <Keywords name="instre1">abstract as async become box break const continue crate do dyn else enum extern false final fn for if impl in let loop macro match mod move mut override priv pub ref return self static struct super trait true try type typeof unsafe unsized use virtual where while yield</Keywords>
            <Keywords name="instre2">bool char f32 f64 i128 i16 i32 i64 i8 isize str u128 u16 u32 u64 u8 usize</Keywords>
            <Keywords name="type1">Self</Keywords>
            <Keywords name="type2"></Keywords>
            <Keywords name="type3"></Keywords>
            <Keywords name="type4"></Keywords>
            <Keywords name="type5"></Keywords>

You also mentioned adding log files, where are they located??

**Comment by Ekopalypse** [comment]
When you open a Rust file, does Npp set the Rust language, i.e. does it show `Rust file` in the status bar? Are you using the default theme or a different one? If you are using a different one, you need to check if `Settings->Style Configurator->Language` has a `Rust` language and if the different styles are defined, i.e. if they use different colors and not one color definition for all.

If you open `Plugins->NppLspClient->Toggle console` you will see the log messages. 
Copy and paste them here.

**Comment by MurraySobol** [comment]
Hi:

      See answers below.

Murray

________________________________

**Comment by Ekopalypse** [comment]
Your `autostart` is set to `false` which means you have to start the language server manually via plugins->NppLspClient->Start server for current language.

**Comment by MurraySobol** [comment]
Hi (again!):

      I started the server:


      and got this message:
[1733731961914060] start_client - "C:\\rust-analyzer-x86_64-pc-windows-msvc\\rust-analyzer.exe"
[1733731961918032]  rust language server started

Failed to load workspaces.
Failed to load workspaces.


[1733732071391828] start_client - no configured server found for normal text

      Am I missing (again) 1 more setting??

Thanks
Murray



________________________________

**Comment by Ekopalypse** [comment]
You need to use Npps "Folder as Workspace dialog (FAW)". I.e

![{E3201E73-4AD2-4D22-8EF2-A4A1ED7784E5}](https://github.com/user-attachments/assets/3f48cb8e-8bc1-4ea2-be0b-ed99a3244133)

Lsp servers such as rust-analyzer need to know where the root directory of the current project is located, in the case of rust-analyzer this is the directory where the cargo.toml file is located, so that they know what and how something needs to be compiled.
This information is provided with the initialization request sent by the client by determining the root nodes from the FAW dialog.

**Comment by Ekopalypse** [comment]
No, that was just an example. You need to add your project directory to the FAW, the directory that contains the root cargo.toml file of your project.

Another example:
![{CAE433E7-B83B-45D1-A287-3104CCE95D64}](https://github.com/user-attachments/assets/791d84e2-f198-4440-aa98-662f812da5e3)

To create a rust binary (aka exe) you call `cargo new NAME_OF_YOUR_PROJECT`
In my case I named it `an_example`

Then you add this folder, in my case it is `D:\temp\deleteme\test\an_example` to the FAW.
